x/httpproxy/proxy_handler.go

```diff
@@ -32,6 +34,17 @@ func (h *proxyHandler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http.R
 		h.connectHandler.ServeHTTP(proxyResp, proxyReq)
 		return
 	}
+	if strings.HasPrefix(proxyReq.URL.Path, "http://") || strings.HasPrefix(proxyReq.URL.Path, "https://") {
```

Comment:
See the forward proxy.
Perhaps the logic should be `proxyReq.URL.Host == ""`, meaning the URL is not absolute.

Also, use the scheme field of the URL, not HasPrefix.